pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -58,14 +57,64 @@ func (e *OpenTsdbExecutor) Execute(ctx context.Context, queries tsdb.QuerySlice,
 
 	tsdbQuery.Start = queryContext.TimeRange.GetFromAsMsEpoch()
 	tsdbQuery.End = queryContext.TimeRange.GetToAsMsEpoch()
+	tsdbQuery.Queries = make([]map[string]interface{}, len(queries))
 
-	for _, query := range queries {
-		tsdbQuery.Queries = []OpenTsdbMetric{
-			OpenTsdbMetric{
-				Metric:     query.Model.Get("metric").MustString(),
-				Aggregator: query.Model.Get("aggregator").MustString(),
-			},
+  for i := 0; i < len(queries); i++ {
+
+		metric := make(map[string]interface{})
+
+		// Setting metric and aggregator		
+		metric["metric"] = queries[i].Model.Get("metric").MustString()
+		metric["aggregator"] = queries[i].Model.Get("aggregator").MustString()
+
+		// Setting downsampling options
+		disableDownsampling := queries[i].Model.Get("disableDownsampling").MustBool()
+		if !disableDownsampling {
+			downsampleInterval := queries[i].Model.Get("downsampleInterval").MustString()
+			if downsampleInterval == "" {
+				downsampleInterval = "1m"  //default value for blank
+			}
+			downsample :=  downsampleInterval + "-" + queries[i].Model.Get("downsampleAggregator").MustString()
+			if queries[i].Model.Get("downsampleFillPolicy").MustString() != "none" {
+				metric["downsample"] = downsample + "-" + queries[i].Model.Get("downsampleFillPolicy").MustString()
+			}
 		}
+
+		// Setting rate options
+		if queries[i].Model.Get("shouldComputeRate").MustBool() {
```

Comment:
some go tests for this would be nice, see the Prometheus implementation for reference :)